poppy/wfe.py

```diff
@@ -20,16 +21,72 @@
 
 __all__ = ['WavefrontError', 'ParameterizedWFE', 'ZernikeWFE']
 
+def _accept_wavefront_or_meters(f):
+    """Decorator that ensures the first positional method argument
+    is a poppy.Wavefront or a floating point number of meters
+    for a wavelength
+    """
+
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if not isinstance(args[1], Wavefront):
+            wave = Wavefront(wavelength=args[1])
+            new_args = (args[0],) + (wave,) + (args[2:])
+            return f(*new_args, **kwargs)
+        else:
+            return f(*args, **kwargs)
+    return wrapper
+
 class WavefrontError(AnalyticOpticalElement):
+    """A base class for different sources of wavefront error
+
+    Analytic optical elements that represent wavefront error should
+    derive from this class and override methods appropriately.
+    Defined to be a pupil-plane optic.
+    """
     def __init__(self, **kwargs):
         super(WavefrontError, self).__init__(planetype=_PUPIL, **kwargs)
 
+    @_accept_wavefront_or_meters
+    def get_opd(self, wave, units='meters'):
+        """Construct the optical path difference array for a wavefront error source
```

Comment:
This reminds me, I have a branch which has refactored most of `optics.py` to use this same approach, `get_opd`/`get_phasor`/ instead of the current `getPhasor` only. I should push that sometime, but I was hoping to leave that for after this release. 

Totally fine to include this here for now, of course. I think the generic `getPhasor` here will be redundant with the one on base `AnalyticOpticalElement` once that other branch is merged, but we can clean this up later on.